mountpoint-s3/src/prefetch.rs

```diff
@@ -48,8 +48,17 @@ pub struct PrefetcherConfig {
 
 impl Default for PrefetcherConfig {
     fn default() -> Self {
+        #[allow(clippy::identity_op)]
         Self {
-            first_request_size: 256 * 1024,
+            // This is a weird looking number! We really want our first request size to be 1MiB,
+            // which is a common IO size. But Linux's readahead will try to read an extra 128k on on
+            // top of a 1MiB read, which we'd have to wait for a second request to service. Because
+            // FUSE doesn't know the difference between regular reads and readahead reads, it will
+            // send us a READ request for that 128k, so we'll have to block waiting for it even if
+            // the application doesn't want it. This is all in the noise for sequential IO, but
+            // waiting for the readahead hurts random IO. So we add 128k to the first request size
+            // to avoid the latency hit of the second request.
+            first_request_size: 1 * 1024 * 1024 + 128 * 1024,
```

Comment:
This readahead value, is it the same across linux distributions?

Reply:
No, it's baked into the kernel: https://elixir.bootlin.com/linux/latest/source/include/linux/pagemap.h#L1185 It's been 128k basically forever, although FUSE didn't always support > 4k reads.